From e3fe550664952493256a0f664a4672d2263a0aee Mon Sep 17 00:00:00 2001
From: wbs306 <wbs306@gmail.com>
Date: Thu, 27 Oct 2022 20:58:17 +0800
Subject: [PATCH 3/6] Camera: Expose aux camera if packagename is null

* fixes ultra-wide and macro lens feature on some devices

Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 core/java/android/hardware/Camera.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/java/android/hardware/Camera.java b/core/java/android/hardware/Camera.java
index e986320b4a9f..5155b30da4cd 100644
--- a/core/java/android/hardware/Camera.java
+++ b/core/java/android/hardware/Camera.java
@@ -273,6 +273,8 @@ public class Camera {
          * if the package name does not falls in this bucket
          */
         String packageName = ActivityThread.currentOpPackageName();
+    	if (packageName == null)
+    	    return true;
         List<String> packageList = new ArrayList<>(Arrays.asList(
                 SystemProperties.get("vendor.camera.aux.packagelist", ",").split(",")));
         List<String> packageExcludelist = new ArrayList<>(Arrays.asList(
-- 
2.34.1

